src/Item_Cluster.php

```diff
@@ -152,96 +148,40 @@ public static function showItems(Cluster $cluster)
-        $types = $CFG_GLPI['cluster_types'];
-        $translated_types = [];
-        foreach ($types as $type) {
-            $translated_types[$type] = $type::getTypeName(1);
-        }
-        Dropdown::showFromArray(
-            'itemtype',
-            $translated_types,
-            [
-                'display_emptychoice'   => true,
-                'value'                 => $this->fields["itemtype"],
-                'rand'                  => $rand
-            ]
-        );
 
-       //get all used items
+        // get all used items
         $used = [];
         $iterator = $DB->request([
-            'FROM'   => $this->getTable()
+            'SELECT' => ['itemtype', 'items_id'],
+            'FROM'   => static::getTable()
         ]);
         foreach ($iterator as $row) {
             $used [$row['itemtype']][] = $row['items_id'];
         }
 
-        Ajax::updateItemOnSelectEvent(
-            "dropdown_itemtype$rand",
-            "items_id",
-            $CFG_GLPI["root_doc"] . "/ajax/dropdownAllItems.php",
-            [
-                'idtable'   => '__VALUE__',
-                'name'      => 'items_id',
-                'value'     => $this->fields['items_id'],
-                'rand'      => $rand,
-                'used'      => $used
-            ]
-        );
-
-        echo "</td>";
-        echo "<td><label for='dropdown_items_id$rand'>" . _n('Item', 'Items', 1) . "</label></td>";
-        echo "<td id='items_id'>";
-        if (isset($this->fields['itemtype']) && !empty($this->fields['itemtype'])) {
-            $itemtype = $this->fields['itemtype'];
-            $itemtype = new $itemtype();
-            $itemtype::dropdown([
-                'name'   => "items_id",
-                'value'  => $this->fields['items_id'],
-                'rand'   => $rand
-            ]);
-        } else {
-            Dropdown::showFromArray(
-                'items_id',
-                [],
-                [
-                    'display_emptychoice'   => true,
-                    'rand'                  => $rand
-                ]
-            );
-        }
-
-        echo "</td>";
-        echo "</tr>";
-
-        echo "<tr class='tab_bg_1'>";
-        echo "<td><label for='dropdown_clusters_id$rand'>" . Cluster::getTypeName(1) . "</label></td>";
-        echo "<td>";
-        Cluster::dropdown(['value' => $this->fields["clusters_id"], 'rand' => $rand]);
-        echo "</td>";
-        echo "</tr>";
-
-        $this->showFormButtons($options);
-
+        $twig_params = [
+            'item' => $this,
+            'used' => $used,
+        ] + $options;
+
+        // language=Twig
+        echo TemplateRenderer::getInstance()->renderFromStringTemplate(<<<TWIG
+            {% extends 'generic_show_form.html.twig' %}
+            {% import 'components/form/fields_macros.html.twig' as fields %}
+            {% block form_fields %}
+                {{ fields.dropdownItemsFromItemtypes('itemtype', _n('Item', 'Items', 1), {
```

Comment:
`_n('Item', 'Items', 1)` will not be detected by the `xgettext` parser.

Maybe this templace could be moved in a `templates/pages/assets/item_cluster.html.twig` file.